inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -0,0 +1,134 @@
+
+from inginious.frontend.parsable_text import ParsableText
+
+class UserHintManager(object):
+
+    """
+        Manage the user hints information in database, as the hints
+        that was unlocked, it's individual penalty, and cumulative
+        penalty, to be applied in the student's final grade for the
+        respective task.
+    """
+
+    def __init__(self, username, task_id, database):
+
+        self._username = username
+        self._task_id = task_id
+        self._database = database
+
+    def check_locked_hint_status(self, hints):
+
+        """ Method to check each hint status, and return the content
+            if it is allowed
+        """
+        user_hints = self._database.user_hints.find_one({"taskid":  self._task_id,
+                                            "username":  self._username})
+        if user_hints is None:
+            user_hints = self.create_hint_list_for_user()
+            return 200, ""
+
+        self.update_allowed_hints_list(user_hints, hints)
+
+        to_show_hints = {}
+        allowed_hints = user_hints["allowedHints"]
+        for key in hints:
+            to_show_hint_content = {
+                "title": hints[key]["title"],
+                "content": None,
+                "penalty": hints[key]["penalty"],
+                "allowed_to_see": False,
+            }
+            if hints[key]["id"] in [hint["id"] for i, hint in enumerate(allowed_hints)]:
```

Comment:
This is very time consuming, why don't you just create like a set of ids before the `for`? you are creating the same list for  each iteration. Also, why do you use the `enumerate` if you are not using the index anyways?